pkg/asset/installconfig/azure/client.go

```diff
@@ -152,3 +154,29 @@ func (c *Client) getProvidersClient(ctx context.Context) (azres.ProvidersClient,
 	client.Authorizer = c.ssn.Authorizer
 	return client, nil
 }
+
+// GetResourceSku checks if the diskType exists for the given region.
+func (c *Client) GetResourceSku(ctx context.Context, region string, diskType string) (*azsku.ResourceSku, error) {
```

Comment:
the function name and the work don't match. I think we should use a more appropriate name